src/test/java/net/openhft/chronicle/queue/InternalAppenderWriteBytesTest.java

```diff
@@ -113,17 +153,64 @@ public void testJumpingAMessageThrowsAIllegalStateException() {
                 .rollCycle(MINUTELY)
                 .timeProvider(() -> 0).build();
 
-             ExcerptAppender appender = q.acquireAppender()) {
-            appender.writeText("hello");
+            ExcerptAppender appender = q.acquireAppender()) {
             appender.writeText("hello");
+            appender.writeText("hello2");
             try (final DocumentContext dc = appender.writingDocument()) {
                 dc.wire().bytes().writeLong(1);
             }
 
             final long l = appender.lastIndexAppended();
             final RollCycle rollCycle = q.rollCycle();
-            final long index = rollCycle.toIndex(rollCycle.toCycle(l) + 1, 1);
+            final int currentCycle = rollCycle.toCycle(l);
+            // try to write to next roll cycle and write at seqnum 1 (but miss the 0th seqnum of that roll cycle)
+            final long index = rollCycle.toIndex(currentCycle + 1, 1);
             ((InternalAppender) appender).writeBytes(index, Bytes.from("text"));
         }
     }
+
+    @Test
+    public void appendToPreviousCycle() {
+        @NotNull Bytes<byte[]> test = Bytes.from("hello world");
+        @NotNull Bytes<byte[]> test1 = Bytes.from("hello world again cycle1");
+        @NotNull Bytes<byte[]> test2 = Bytes.from("hello world cycle2");
+        Bytes result = Bytes.elasticHeapByteBuffer();
+        SetTimeProvider timeProvider = new SetTimeProvider();
+        try (SingleChronicleQueue q = SingleChronicleQueueBuilder.binary(getTmpDir()).timeProvider(timeProvider).rollCycle(MINUTELY).build()) {
+            ExcerptAppender appender = q.acquireAppender();
+            appender.writeBytes(test);
+            long nextIndexInFirstCycle = appender.lastIndexAppended() + 1;
+            int firstCycle = q.rollCycle().toCycle(nextIndexInFirstCycle);
+
+            timeProvider.advanceMillis(TimeUnit.SECONDS.toMillis(65));
+            appender.writeBytes(test2);
+
+            Assert.assertTrue(hasEOF(q, firstCycle));
+            // here we try and write to previous cycle file. We will overwrite the EOF in doing so
+            expectException("Incomplete header found at pos: 33048: c0000000, overwriting");
+            ((InternalAppender) appender).writeBytes(nextIndexInFirstCycle, test1);
```

Comment:
this does work currently i.e. we can add to the previous cycle. If we want to support this as a use case we will need to re-write the EOF at the end, if we overwrote it

Reply:
this feature could be useful to keep in case e.g. the pretoucher has rolled the queue already. Although if the queue has rolled and there is data in the next cycle then we probably don't want to allow this

Reply:
We are happy with current behaviour. javadoc now reflects this test